runtime/jvmti/jvmtiClass.c

```diff
@@ -1278,6 +1278,9 @@ redefineClassesCommon(jvmtiEnv* env,
 			/* Notify the JIT about redefined classes */
 			jitClassRedefineEvent(currentThread, &jitEventData, FALSE, FALSE);
 
+			/* Clear/suspend all breakpoints in the classes being replaced */
+			clearBreakpointsInClasses(currentThread, classPairs);
```

Comment:
For consistency, this should probably happen first, as it does in the else case below.

Reply:
Or better yet, call it before the extensions check.

Reply:
If it is called before `fixJNIRefs` the test fails. Specifically, the `if (J9_CLASS_FROM_METHOD(((J9JNIMethodID *) agentBreakpoint->method)->method) == originalRAMClass) {` will fail because the class in the breakpointed method will point to the new class instead of the original class. 

I believe the reason for this is that the class swap is not fully complete until all the fixups (like JNI methodIDs) are done. Whereas in the extended (non-fast) HCR case we are not swapping but instead creating a new version.